### PR TITLE
[SPARK-34774][BUILD] Ensure change-scala-version.sh update scala.version in parent POM correctly

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
After SPARK-34507,  execute` change-scala-version.sh` script will update `scala.version` in parent pom, but if we execute the following commands in order：

```
dev/change-scala-version.sh 2.13
dev/change-scala-version.sh 2.12
git status
```

there will generate git diff as follow:

```
diff --git a/pom.xml b/pom.xml
index ddc4ce2f68..f43d8c8f78 100644
--- a/pom.xml
+++ b/pom.xml
@@ -162,7 +162,7 @@
     <commons.math3.version>3.4.1</commons.math3.version>
     <!-- managed up from 3.2.1 for SPARK-11652 -->
     <commons.collections.version>3.2.2</commons.collections.version>
-    <scala.version>2.12.10</scala.version>
+    <scala.version>2.13.5</scala.version>
     <scala.binary.version>2.12</scala.binary.version>
     <scalatest-maven-plugin.version>2.0.0</scalatest-maven-plugin.version>
     <scalafmt.parameters>--test</scalafmt.parameters>
```

seem 'scala.version' property was not update correctly.

So this pr add an extra 'scala.version' to scala-2.12 profile to ensure change-scala-version.sh can update the public `scala.version` property correctly. 

### Why are the changes needed?
Bug fix.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
**Manual test**

Execute the following commands in order：

```
dev/change-scala-version.sh 2.13
dev/change-scala-version.sh 2.12
git status
```

**Before**

```
diff --git a/pom.xml b/pom.xml
index ddc4ce2f68..f43d8c8f78 100644
--- a/pom.xml
+++ b/pom.xml
@@ -162,7 +162,7 @@
     <commons.math3.version>3.4.1</commons.math3.version>
     <!-- managed up from 3.2.1 for SPARK-11652 -->
     <commons.collections.version>3.2.2</commons.collections.version>
-    <scala.version>2.12.10</scala.version>
+    <scala.version>2.13.5</scala.version>
     <scala.binary.version>2.12</scala.binary.version>
     <scalatest-maven-plugin.version>2.0.0</scalatest-maven-plugin.version>
     <scalafmt.parameters>--test</scalafmt.parameters>
```

**After**

No git diff.

### CODE REVIEW

**Comment by LuciferYang** [comment]
cc @srowen @HyukjinKwon

File: pom.xml [hunk 1]
```diff
@@ -3300,6 +3300,13 @@
 
     <profile>
       <id>scala-2.12</id>
+      <properties>
+        <!-- 
+         SPARK-34774 Add this property to ensure change-scala-version.sh can replace the public `scala.version`
+         property correctly. 
+        -->
+        <scala.version>2.12.10</scala.version>
```
Comment (LuciferYang) [inline pom.xml:3308]: This change will produce two`< scala.version >2.12.10</ scala.version >`, Should we extract a common property like `< scala.default.version >2.12.10</scala.default.version >` and reference it here as `<scala.version>${scala.default.version}</scala.version>` ?

**Comment by sarutak** [comment]
Hmm, this seems to duplicate #31863.

**Comment by LuciferYang** [comment]
> Hmm, this seems to duplicate #31863.

@sarutak It seems that we found the same problem almost at the same time :) 

But maybe add a `scala.version`  property to the scala-2.12  profile will be able to retain the fix for SPARK-34507

**Comment by sarutak** [comment]
O.K, this solution can be more prominent.

**Comment by srowen** [comment]
I'm surprised if it can't be evaluated without being in a profile?

**Comment by LuciferYang** [comment]
> I'm surprised if it can't be evaluated without being in a profile?

Yes, at least the above test method will trigger this problem, @srowen I'm not sure if there's a better way to fix it

**Comment by srowen** [comment]
Fix seems fine is so FWIW

**Comment by HyukjinKwon** [comment]
I'm okay with this.

**Comment by srowen** [comment]
(Jenkins won't test this anyway, note)
Merged to master / 3.1 / 3.0.
@LuciferYang do you want to make a similar change for 2.x? I can do it too.

**Comment by LuciferYang** [comment]
> @LuciferYang do you want to make a similar change for 2.x? I can do it too.


@srowen OK, I will do this later ~

**Comment by LuciferYang** [comment]
> @LuciferYang do you want to make a similar change for 2.x? I can do it too.

@srowen for  branch-2.4: https://github.com/apache/spark/pull/31893

**Pull request closed without merging**